github/repos_contents.go

```diff
@@ -53,12 +54,26 @@ type RepositoryContentFileOptions struct {
 	Committer *CommitAuthor `json:"committer,omitempty"`
 }
 
-// RepositoryContentGetOptions represents an optional ref parameter, which can be a SHA,
+type RepositoryContentMediaType uint8
+
+const (
+	Object RepositoryContentMediaType = iota
+	Raw
+	HTML
+)
+
+// RepositoryContentParameters represents an optional ref parameter, which can be a SHA,
 // branch, or tag
-type RepositoryContentGetOptions struct {
+type RepositoryContentParameters struct {
 	Ref string `url:"ref,omitempty"`
 }
 
+// RepositoryContentGetOptions represents the optional custom media type and ref parameters.
+type RepositoryContentGetOptions struct {
+	Params    RepositoryContentParameters
+	MediaType RepositoryContentMediaType
+}
```

Comment:
@kshitij10496 how about having this instead?

```
type RepositoryContentGetOptions struct {
 	Ref       string `url:"ref,omitempty"`
 	MediaType RepositoryContentMediaType
 }
```

This will ensure the API's backward compatibility and any new options (in future) can still be added to `RepositoryContentGetOptions` struct instead of `RepositoryContentParameters`. What do you think?

Reply:
I see that it was suggested in your approach #2 on the issue. However, I don't understand the cons of this approach.

Reply:
The issues we might face if on proceeding with Approach #02:
1. The `addOptions` function expects a value of struct type with url tags. Thus, we will have to declare another type to handle this for us.
2. As a result of the above impediment, adding a new parameter to the `RepositoryContentsGetOptions` in the future will require a lot of corresponding changes to be made.

At a fundamental level, is it a good practice to encapsulate the request parameters and the request headers into a single type value as raw data rather than composing them from simpler structs?